### PR TITLE
streams-intro doc cleanups

### PR DESCRIPTION
**Pull request by enjoy-binbin**
1. Fixed some typos.
2. Use `` instead of **** for commands, so that we can
generate a hyperlink to jump to the command page.

### CODE REVIEW

File: topics/streams-intro.md [hunk 1]
```diff
@@ -8,18 +8,18 @@ Consumer groups were initially introduced by the popular messaging system Kafka
...
 
 ```
 > XADD mystream * sensor-id 1234 temperature 19.8
 1518951480106-0
 ```
 
-The above call to the **XADD** command adds an entry `sensor-id: 1234, temperature: 19.8` to the stream at key `mystream`, using an auto-generated entry ID, which is the one returned by the command, specifically `1518951480106-0`. It gets as its first argument the key name `mystream`, the second argument is the entry ID that identifies every entry inside a stream. However, in this case, we passed `*` because we want the server to generate a new ID for us. Every new ID will be monotonically increasing, so in more simple terms, every new entry added will have a higher ID compared to all the past entries. Auto-generation of IDs by the server is almost always what you want, and the reasons for specifying an ID explicitly are very rare. We'll talk more about this later. The fact that each Stream entry has an ID is another similarity with log files, where line numbers, or the byte offset inside the file, can be used in order to identify a given entry. Returning back at our **XADD** example, after the key name and ID, the next arguments are the field-value pairs composing our stream entry.
```
Comment (enjoy-binbin) [inline topics/streams-intro.md:20]: diff:
```
- **XADD** command adds an entry
+ `XADD` command adds an entry

- our **XADD** example
+ our `XADD` example
```

File: topics/streams-intro.md [hunk 2]
```diff
@@ -413,15 +413,15 @@ In its simplest form, the command is called with two arguments, which are the na
...
 When called in this way, the command outputs the total number of pending messages in the consumer group (two in this case), the lower and higher message ID among the pending messages, and finally a list of consumers and the number of pending messages they have.
-We have only Bob with two pending messages because the single message that Alice requested was acknowledged using **XACK**.
+We have only Bob with two pending messages because the single message that Alice requested was acknowledged using `XACK`.
 
-We can ask for more information by giving more arguments to **XPENDING**, because the full command signature is the following:
+We can ask for more information by giving more arguments to `XPENDING`, because the full command signature is the following:
 
 ```
-XPENDING <key> <groupname> [<start-id> <end-id> <count> [<consumer-name>]]
+XPENDING <key> <groupname> [[IDLE <min-idle-time>] <start-id> <end-id> <count> [<consumer-name>]]
```
Comment (enjoy-binbin) [inline topics/streams-intro.md:421]: full syntax: https://redis.io/commands/XPENDING

File: topics/streams-intro.md [hunk 3]
```diff
@@ -595,11 +595,15 @@ In case you do not remember the syntax of the command, just ask the command itse
 
 ```
 > XINFO HELP
-1) XINFO <subcommand> arg arg ... arg. Subcommands are:
-2) CONSUMERS <key> <groupname>  -- Show consumer groups of group <groupname>.
-3) GROUPS <key>                 -- Show the stream consumer groups.
-4) STREAM <key>                 -- Show information about the stream.
-5) HELP                         -- Print this help.
+1) XINFO <subcommand> [<arg> [value] [opt] ...]. Subcommands are:
```
Comment (enjoy-binbin) [inline topics/streams-intro.md:598]: This is now (unstable branch), current output

**Review by itamarhaber — APPROVED** [review]
Thanks @enjoy-binbin

**Comment by itamarhaber** [comment]
@banker have you also made changes to this? can this be merged?

File: topics/streams-intro.md [hunk 4]
```diff
@@ -517,15 +517,15 @@ That doesn't mean that there are no new idle pending messages, so the process co
...
+When there are failures, it is normal that messages will be delivered multiple times, but eventually they usually get processed and acknowledged. However there might be a problem processing some specific message, because it is corrupted or crafted in a way that triggers a bug in the processing code. In such a case what happens is that consumers will continuously fail to process this particular message. Because we have the counter of the delivery attempts, we can use that counter to detect messages that for some reasons are not processable. So once the deliveries counter reaches a given large number that you chose, it is probably wiser to put such messages in another stream and send a notification to the system administrator. This is basically the way that Redis Streams implements the *dead letter* concept.
```
Comment (zuiderkwast) [inline topics/streams-intro.md:522]: "for some reason" is better here. It's an idiom meaning that we know that the messages are not processable but we don't know why. Probably it doesn't matter why. (https://www.merriam-webster.com/dictionary/for%20some%20reason, https://www.lexico.com/definition/for_some_reason)

**Review by zuiderkwast — COMMENTED** [review]
Thanks! Most of the changes are good (formatting). I have commented where I don't agree.

**Comment by banker** [comment]
@itamarhaber I haven't made any changes, so go ahead and merge.

File: topics/streams-intro.md [hunk 5]
```diff
@@ -682,7 +686,7 @@ Another special ID is `>`, that is a special meaning only related to consumer gr
 
 Finally the special ID `*`, that can be used only with the `XADD` command, means to auto select an ID for us for the new entry.
 
-So we have `-`, `+`, `$`, `>` and `*`, and all have a different meaning, and most of the times, can be used in different contexts.
+So we have `-`, `+`, `$`, `>` and `*`, and all have a different meaning, and most of the time, can be used in different contexts.
```
[the remark this answers is not in the document]
Comment (enjoy-binbin) [reply topics/streams-intro.md]: ohh, i keep it

File: topics/streams-intro.md (hunk 4 above)
Comment (enjoy-binbin) [reply topics/streams-intro.md]: got it!

File: topics/streams-intro.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (zuiderkwast) [reply topics/streams-intro.md]: Sorry, I mixed up the diff in this one! :confounded: You are correct here. You have changed to "most of the time" which is correct.

[the remark this answers is not in the document]
Comment (zuiderkwast) [reply topics/streams-intro.md:689]: OK, unreverted again. :grinning:

**Comment by zuiderkwast** [comment]
Thanks @enjoy-binbin and sorry for the confusion!

**Pull request merged**